includes/upload_page.form.inc

```diff
@@ -76,10 +82,14 @@ function islandora_paged_content_upload_page_form_submit(array $form, array &$fo
   }
   $object->label = $label;
 
+  // Add the OCR creation flags before the datastream is updated.
+  if (module_exists('islandora_ocr')) {
+    module_load_include('inc', 'islandora_ocr', 'includes/derivatives');
+    islandora_ocr_set_generating_rels_ext_statements($object, !$form_state['values']['ignore_ocr'], !$form_state['values']['ignore_hocr']);
+  }
   islandora_paged_content_update_datastream($object, drupal_realpath($file->uri), 'OBJ', NULL, NULL, 'M', FALSE);
   // Update RELS-EXT properties, page/sequence/etc, and append the page at the
   // end of the book.
-  $rels_ext = $object->relationships;
   if (isset($form_state['values']['language'])) {
     islandora_paged_content_set_relationship($rels_ext, ISLANDORA_RELS_EXT_URI, 'hasLanguage', $form_state['values']['language'], TRUE);
```

Comment:
Same, hasLanguage predicate handled differently, maybe should go into  islandora_ocr_set_generating_rels_ext_statements

Reply:
Handling this more correctly now!